autokeras/hypermodel/preprocessor.py

```diff
@@ -673,3 +676,233 @@ def set_config(self, config):
         self.gaussian_noise = config['gaussian_noise']
         self.seed = config['seed']
         self.shape = config['shape']
+
+
+def return_zero():
+    return 0
+
+
+class FeatureEngineering(Preprocessor):
+    """A preprocessor block does feature engineering for the data.
+
+    # Arguments
+        column_types: A list of strings. The length of the list should be the same
+            as the number of columns of the data. The strings in the list are
+            specifying the types of the columns. They should either be 'numerical'
+            or 'categorical'.
+        max_columns: Int. The maximum number of columns after feature engineering.
```

Comment:
The meaning of this argument isn't entirely clear. Is this about feature selection?